core/roboconf-messaging-http/src/test/java/net/roboconf/messaging/http/DmClientTest.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2014-2015 Linagora, Université Joseph Fourier, Floralis
+ *
+ * The present code is developed in the scope of the joint LINAGORA -
+ * Université Joseph Fourier - Floralis research program and is designated
+ * as a "Result" pursuant to the terms and conditions of the LINAGORA
+ * - Université Joseph Fourier - Floralis research program. Each copyright
+ * holder of Results enumerated here above fully & independently holds complete
+ * ownership of the complete Intellectual Property rights applicable to the whole
+ * of said Results, and may freely exploit it in any manner which does not infringe
+ * the moral rights of the other copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.roboconf.messaging.http;
+
+import java.util.concurrent.LinkedBlockingQueue;
+
+import junit.framework.Assert;
+import net.roboconf.core.model.beans.Application;
+import net.roboconf.messaging.api.client.ListenerCommand;
+import net.roboconf.messaging.api.messages.Message;
+import net.roboconf.messaging.http.internal.HttpDmClient;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Pierre-Yves Gibello - Linagora
+ */
+public class DmClientTest {
+
+	@BeforeClass
+	public static void startServices() {
+		HttpTestUtils.runWebServer();
+		try {
+			Thread.sleep(500); //TODO: any way to start webserver quicky (and remove this) ??
```

Comment:
Le 07/10/2015 11:27, Vincent Zurczak a écrit :
>
> In 
> core/roboconf-messaging-http/src/test/java/net/roboconf/messaging/http/DmClientTest.java 
> <https://github.com/roboconf/roboconf-platform/pull/459#discussion_r41368961>:
>
> > +import net.roboconf.messaging.http.internal.HttpDmClient;
> > +
> > +import org.junit.AfterClass;
> > +import org.junit.BeforeClass;
> > +import org.junit.Test;
> > +
> > +/**
> > + * @author Pierre-Yves Gibello - Linagora
> > + */
> > +public class DmClientTest {
> > +
> > +	@BeforeClass
> > +	public static void startServices() {
> > +		HttpTestUtils.runWebServer();
> > +		try {
> > +			Thread.sleep(500); //TODO: any way to start webserver quicky (and remove this) ??
>
> It is not a big issue to have such waiting.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/roboconf/roboconf-platform/pull/459/files#r41368961>.
>
No, but it makes tests unpredictable (and causes unwanted build failures 
- eg. in the continuous integration process).